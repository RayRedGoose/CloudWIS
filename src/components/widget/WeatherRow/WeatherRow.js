import React from 'react'
import './WeatherRow.scss'
import WeatherItem from '../WeatherItem/WeatherItem'

const WeatherRow = ({ weatherData, title, type }) => {
  const keys = Object.keys(weatherData).filter(key => key !== 'desc')
  const currentItems = keys.map((key) => {
    const itemDesc = (key === 'temp') ? weatherData['desc'] : key
    return (
      <WeatherItem
        key={key}
        weekDay=''
        typeAbbr={key}
        itemType={itemDesc}
        blockType={type}
        number={parseInt(weatherData[key])} />
    )
  })

  const daysItems = Object.keys(weatherData).map((key) => {
    const weatherInfo = weatherData[key].toString().split('-')
    return (
      <WeatherItem
        key={key}
        weekDay={key}
        typeAbbr='temp'
        itemType={weatherInfo[1]}
        blockType={type}
        number={parseInt(weatherInfo[0])} />
    )
  })

  return (
    <section className="weather-row">
      <h2>{title}</h2>
      <main>
        { (type === 'day') ? currentItems : daysItems }
      </main>
    </section>
  )
}

export default WeatherRow
