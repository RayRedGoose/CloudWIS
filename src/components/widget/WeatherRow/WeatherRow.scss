.weather-row {
  padding: 0 20px;
  margin: 40px 0;
  h2 {
    border-bottom: 2px solid #0F1A20;
    color: #F0EEEB;
    margin-top: 20px;
    padding-bottom: 5px;
    padding-right: 80px;
    width: fit-content;
  }
  main {
    margin: 20px 0;
    display: flex;
    width: 100%;
    overflow-x: scroll;
  }
}
