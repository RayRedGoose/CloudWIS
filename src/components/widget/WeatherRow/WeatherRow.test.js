import React from 'react';
import { shallow } from 'enzyme';
import WeatherRow from './WeatherRow';

it('should match the snapshot if it has day type', () => {
  const mockProps = {
    weatherData: {temp: 15},
    title: 'Today',
    type: 'day'
  }

  const row = shallow(
    <WeatherRow {...mockProps} />
  );

  expect(row).toMatchSnapshot();
});

it('should match the snapshot if it has fiveDays type', () => {
  const mockProps = {
    weatherData: {'Dec,01': 15, 'Dec,02': 25},
    title: '5-Days',
    type: 'fiveDays'
  }

  const row = shallow(
    <WeatherRow {...mockProps} />
  );

  expect(row).toMatchSnapshot();
});
