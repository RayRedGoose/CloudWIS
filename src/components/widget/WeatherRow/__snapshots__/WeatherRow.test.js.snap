// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot if it has day type 1`] = `
<section
  className="weather-row"
>
  <h2>
    Today
  </h2>
  <main>
    <WeatherItem
      blockType="day"
      key="temp"
      number={15}
      typeAbbr="temp"
      weekDay=""
    />
  </main>
</section>
`;

exports[`should match the snapshot if it has fiveDays type 1`] = `
<section
  className="weather-row"
>
  <h2>
    5-Days
  </h2>
  <main>
    <WeatherItem
      blockType="fiveDays"
      key="Dec,01"
      number={15}
      typeAbbr="temp"
      weekDay="Dec,01"
    />
    <WeatherItem
      blockType="fiveDays"
      key="Dec,02"
      number={25}
      typeAbbr="temp"
      weekDay="Dec,02"
    />
  </main>
</section>
`;
